core/trie/node.go

```diff
@@ -92,17 +77,16 @@ func (n *Node) UnmarshalBinary(data []byte) error {
 		return nil
 	}
 
-	stream := bytes.NewReader(data)
-
 	if n.Left == nil {
-		n.Left = new(bitset.BitSet)
-		n.Right = new(bitset.BitSet)
+		n.Left = new(Key)
+		n.Right = new(Key)
 	}
 
-	_, err := n.Left.ReadFrom(stream)
-	if err != nil {
+	if err := n.Left.UnmarshalBinary(data); err != nil {
+		return err
+	}
+	if err := n.Right.UnmarshalBinary(data[n.Left.EncodedLen():]); err != nil {
 		return err
 	}
-	_, err = n.Right.ReadFrom(stream)
-	return err
+	return nil
```

Comment:
maybe simplify to `return n.Right.UnmarshalBinary(...)` ?

Reply:
Thanks, will do.

Reply:
Done